### PR TITLE
CMCL-0000: rotate with follow target

### PR DESCRIPTION
**Pull request by glabute**
### Purpose of this PR

Replaced SameAsFollowTarget with RotateWithFollowTarget (SameAsFollowTarget still exists but is deprecated).
This is just for the clearer name.  Made a new component instead of renaming to avoid a major version change.
Modified the upgrader to handle this change.

### Testing status

- [ ] Added an automated test
- [x] Passed all automated tests
- [x] Manually tested 

### Documentation status

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [x] Commented all public classes, properties, and methods
- [x] Updated user documentation

### Technical risk

low

### CODE REVIEW

File: com.unity.cinemachine/Runtime/Components/CinemachineRotateWithFollowTarget.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /// <summary>
+    /// This is a CinemachineComponent in the Aim section of the component pipeline.
+    /// Its job is to match the orientation of the Follow target.
+    /// </summary>
+    [AddComponentMenu("Cinemachine/Procedural/Rotation Control/Cinemachine Rotate With Follow Target")]
+    [SaveDuringPlay]
+    [DisallowMultipleComponent]
+    [CameraPipeline(CinemachineCore.Stage.Aim)]
+    [HelpURL(Documentation.BaseURL + "manual/CinemachineRotateWithFollowTarget.html")]
+    public class CinemachineRotateWithFollowTarget : CinemachineComponentBase
```
Comment (AntoineCharton) [inline com.unity.cinemachine/Runtime/Components/CinemachineRotateWithFollowTarget.cs:14]: Is this for 3.1?

**Review by AntoineCharton — COMMENTED** [review]
I am not 100% sure if this is intended to be for 3.1. Was this change planned?

File: com.unity.cinemachine/Runtime/Components/CinemachineRotateWithFollowTarget.cs (hunk 1 above)
Comment (glabute) [reply com.unity.cinemachine/Runtime/Components/CinemachineRotateWithFollowTarget.cs:14]: Yes

**Review by AntoineCharton — APPROVED** [review]
Looks good. Will need to make sure API validation doesn't break with the deprecation.

**Pull request merged**